engine/flink/management/src/main/scala/pl/touk/nussknacker/engine/management/FlinkRestManager.scala

```diff
@@ -66,28 +69,45 @@ class FlinkRestManager(
       )
   }
 
-  private def toJobStatus(overview: JobOverview): JobStatus = {
-    import org.apache.flink.api.common.JobStatus
-    JobStatus.valueOf(overview.state)
+  override def getDeploymentStatusesToUpdate: Future[Map[newdeployment.DeploymentId, DeploymentStatus]] = {
+    client.getJobsOverviews()(DataFreshnessPolicy.Fresh).map(_.value).flatMap { jobsOverviews =>
+      jobsOverviews
+        .map { jobOverview =>
+          val status = mapJobStatus(jobOverview)
+          client.getJobConfig(jobOverview.jid).map { jobConfig =>
+            jobConfig.`user-config`
+              .get("deploymentId")
+              .flatMap(_.asString)
+              .flatMap(newdeployment.DeploymentId.fromString)
+              .map(_ -> status)
+          }
+        }
+        .sequence
+        .map(_.flatten.toMap)
+    }
   }
 
   // NOTE: Flink <1.10 compatibility - protected to make it easier to work with Flink 1.9, JobStatus changed package, so we use String in case class
-  protected def mapJobStatus(overview: JobOverview): StateStatus = {
+  protected def mapJobStatus(overview: JobOverview): DeploymentStatus = {
```

Comment:
WDYT about move this method to dedicated companion object?

Reply:
I can't move it. See a NOTE above. Also this methods has another comment in the body, that explain this comment. I don't want to detach them.